### ISSUE TITLE
App crash for iOS 12.5 or lower for Branch 5.1.0

### ISSUE THREAD

**Issue by theamorn** [RodrigoSMarques/flutter_branch_sdk#162]
**Describe the bug**
The app is crashed for iOS 12.5 and lower with Flutter 3.0.1 (On production I use 3.0.1)

**To Reproduce**
1. Create a simple app with Flutter 3.0.3 (for development, I upgraded to 3.0.3 already)
2. install Branch SDK 5.1.0
3. Run on iOS 12.4.8, iPhone 6 Plus
4. App will terminate and quit while on the white screen the first time it's started
5. Rollback to Branch SDK 4.0.0 and it's working fine

**Expected behavior**
It should be working on supported OS, it's working fine on iOS 13, 14,15 and 16

**Screenshots**
```
default	19:00:45.693336+0700	assertiond	[Runner:593] Invalidating...
error	19:00:45.693748+0700	assertiond	[Runner:593] SyscallError: setpriority(PRIO_DARWIN_ROLE, 593, 3): No such process
default	19:00:45.799233+0700	assertiond	[Runner:593] Got exit context: <BKSProcessExitContext: 0x1007283d0; reason: (none)>
default	19:00:45.800275+0700	SpringBoard	<FBApplicationProcess: 0x125548210; Runner (com.kbtg.neobank); pid: 593> assertiond says the process actually exited with context: <BKSProcessExitContext: 0x283205800; reason: (none)>
default	19:00:45.801063+0700	SpringBoard	<FBApplicationProcess: 0x125548210; Runner (com.kbtg.neobank); pid: 593> exited.
default	19:00:45.813798+0700	SpringBoard	Removing: <FBApplicationProcess: 0x125548210; Runner (com.kbtg.neobank); pid: -1>
default	19:00:45.921406+0700	assertiond	[Runner:593] Terminating because the job-submitter has disconnected.
default	19:00:45.921709+0700	assertiond	[Runner:593] Deleted launchd job with label: UIKitApplication:com.kbtg.neobank[0x1915][62]
default	19:00:45.921948+0700	assertiond	[Runner:593] Invalidation complete.
default	19:00:45.974163+0700	assertiond	[Runner:593] Removing client: <BKProcessInfoServerClient: 0x1007246a0; pid: 52>
default	19:00:45.974251+0700	assertiond	[Runner:593] No clients remain.
default	19:00:45.984472+0700	SpringBoard	Process exited: <FBApplicationProcess: 0x125548210; Runner (com.kbtg.neobank); pid: -1> -> <FBApplicationProcessExitContext: 0x283f0c120; exitReason: (none); terminationReason: (none)> {
    stateAtExit = <FBProcessState: 0x28311e880; pid: 593; taskState: Not Running; visibility: Unknown>;
}
default	19:00:46.037354+0700	ReportCrash	Process:             Runner [593]
Path:                /private/var/containers/Bundle/Application/F4976403-0679-4CAE-9465-97FAA8A47E23/Runner.app/Runner
OS Version:          iPhone OS 12.4.8 (16G201)
default	19:00:46.037568+0700	ReportCrash	Exception Type:  EXC_CRASH (SIGABRT)
Exception Codes: 0x0000000000000000, 0x0000000000000000
Exception Note:  EXC_CORPSE_NOTIFY
Termination Description: DYLD, Library not loaded: /System/Library/Frameworks/LinkPresentation.framework/LinkPresentation | Referenced from: /private/var/containers/Bundle/Application/F4976403-0679-4CAE-9465-97FAA8A47E23/Runner.app/Frameworks/Branch.framework/Branch | Reason: image not found
Triggered by Thread:  0
```

<img width="80%" alt="Screen Shot 2565-06-23 at 19 03 03" src="https://user-images.githubusercontent.com/6716483/175294304-912cf45a-1c7c-42b9-8553-8f5270a14666.png">
<img width="80%" alt="Screen Shot 2565-06-23 at 19 03 11" src="https://user-images.githubusercontent.com/6716483/175294323-af0950ef-8c44-46ca-a6c4-124d1e968e5f.png">

it seems this line has a problem 
`Termination Description: DYLD, Library not loaded: /System/Library/Frameworks/LinkPresentation.framework/LinkPresentation | Referenced from: /private/var/containers/Bundle/Application/F4976403-0679-4CAE-9465-97FAA8A47E23/Runner.app/Frameworks/Branch.framework/Branch | Reason: image not found`

**Desktop (please complete the following information):**
[✓] Flutter (Channel stable, 3.0.3, on macOS 12.4 21F79 darwin-arm, locale en-TH)
    • Flutter version 3.0.3 at /Users/amorn/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 676cefaaff (18 hours ago), 2022-06-22 11:34:49 -0700
    • Engine revision ffe7b86a1e
    • Dart version 2.17.5
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /Users/amorn/Library/Android/sdk
    • Platform android-32, build-tools 32.1.0-rc1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.68.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.42.0

[✓] Connected device (5 available)
    • Redmi Note 8 (mobile) • 16368138                                 • android-arm64  • Android 11 (API 30)
    • CPH1803 (mobile)      • cfb7fef0                                 • android-arm64  • Android 8.1.0 (API 27)
    • iPhone (mobile)       • e9db267842dccbc84914dcd56558a478f9730677 • ios            • iOS 12.4.8 16G201
    • macOS (desktop)       • macos                                    • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)          • chrome                                   • web-javascript • Google Chrome 103.0.5060.53
    ! Error: iPhone is busy: Fetching debug symbols for iPhone. Xcode will continue when iPhone is finished. (code -10)

[✓] HTTP Host Availability
    • All required HTTP hosts are available

**Smartphone (please complete the following information):**
 iOS 12.4.8 iPhone 6Plus

**Comment by theamorn** [comment]
I believe that it should be related to this issue in Flutter
https://github.com/fluttercommunity/plus_plugins/pull/224

so it needs to change to weak_framework instead for older devices

**Comment by RodrigoSMarques** [comment]
Hi @theamorn.

This is not related to anything in the package, but a failure to integrate pods through cocoapods.

Try doing the following:

- run `flutter clean` command
- go to your project's `ios folder and run the `pod install` command

Check if there is an error.

I created a new project, added the package.

When I run the `pod install`, I get an error.

I edited the `Podfile` file to include the instructions below:

```swift
...
flutter_ios_podfile_setup

target 'Runner' do
  use_frameworks!  #<--- added here 
  use_modular_headers!  #<--- added here 

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
  end
end
```

**Comment by RodrigoSMarques** [comment]
> I believe that it should be related to this issue in Flutter [fluttercommunity/plus_plugins#224](https://github.com/fluttercommunity/plus_plugins/pull/224)
> 
> so it needs to change to weak_framework instead for older devices

Hi @theamorn 

A new version of the Branch SDK for iOS has been released and the `LinkPresentation.framework` issue has been fixed.

A new version of the `flutter_branch_sdk` package will be released in the next one and it will already include the new version.

More information here:

[https://github.com/BranchMetrics/ios-branch-deep-linking-attribution/pull/1181](https://github.com/BranchMetrics/ios-branch-deep-linking-attribution/pull/1181)
[https://github.com/BranchMetrics/ios-branch-deep-linking-attribution/releases/tag/1.43.1](https://github.com/BranchMetrics/ios-branch-deep-linking-attribution/releases/tag/1.43.1)

**Comment by RodrigoSMarques** [comment]
Fix on Release 6.0.0
